### PR TITLE
tests: only test changed packages

### PR DESCRIPTION
**Pull request by defaude**
Relates to #1193

### CODE REVIEW

**Review by appgurueu — CHANGES_REQUESTED** [review]
This is akin to outcommenting the test.

Project Euler solutions can be expected to take some time (should be < one minute though, ideally the algo should be optimized).

If you do not want to run this test, simply exclude the directory when running tests using Jest.

**Comment by defaude** [comment]
However, having this long-running tests clogs up everyone's workflow and not "just" the CI, therefore I recommend to **not** just uncommenting the test but rather explicitly skipping it. This way, it's still visible in the console. Finding a solution to the runtime behavior is step 2 - but meanwhile, everyone else can continue working without that delay.

What do you think?

**Review by appgurueu — DISMISSED** [review]

**Comment by appgurueu** [comment]
We'll have to leave the issue open to find a proper fix eventually however.

**Comment by defaude** [comment]
Yup! 👍  I mereley mentioned the issue no. in the commit message but didn't _close_ it.

**Comment by raklaptudirm** [comment]
Ideally, we would want to run the tests in the changed directories only.

**Comment by defaude** [comment]
Jest has an `onlyChanged` option that I've set up in the pre-commit hook. If that's what you like, we could also switch to that behavior for the CI, as well.

**Comment by raklaptudirm** [comment]
That could be great!

**Comment by defaude** [comment]
@raklaptudirm I've updated the CI config, as well 👍

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -107,6 +108,12 @@ You can also start Jest in "watch" mode:
 npm test -- --watchAll
 ```
 
+We also prepared a helper script that runs tests only for changed files:
+
+```shell
+npm run test-watch
```
Comment (appgurueu) [inline CONTRIBUTING.md:114]: Shouldn't this be `npm run test-changed`?

Comment (defaude) [reply CONTRIBUTING.md:114]: Of course (the day was too long, sorry... 😅) I'll update my PR immediately.

**Review by appgurueu — DISMISSED** [review]
Thank you for your contribution!

**Comment by raklaptudirm** [comment]
The pr looks fine, but I had an idea. I think we would want to run all the tests in the master branch, but run tests on only the changed files in the prs. What do you all think?

**Comment by defaude** [comment]
I've updated the Workflow accordingly - please take a look 😀

**Comment by raklaptudirm** [comment]
> I guess I could set that up with GitHub Actions, yeah. But as long as #1193 is not solved, the `master` build will take forever to complete.

The master build should not be an issue as no one will need to wait for that to complete.

File: .github/workflows/Ci.yml [hunk 2]
```diff
@@ -20,8 +21,13 @@ jobs:
...
 
-      - name: 🧪 Run tests
-        run: npm test
+      - name: 🧪 Run all tests
+        if: ${{ github.event_name == 'push' }}
+        run: npm run test
+
+      - name: 🧪 Run tests for changed files only
+        if: ${{ github.event_name == 'pull_request' }}
+        run: npm run test
```
Comment (raklaptudirm) [inline .github/workflows/Ci.yml:30]: This will be test-changed i guess

Comment (defaude) [reply .github/workflows/Ci.yml]: You're absolutely right....... 😨

**Review by appgurueu — APPROVED** [review]

**Pull request merged**